### PR TITLE
Rails Capstone

### PR DESCRIPTION
**Pull request by AbigiyaTY**
In this project I:

**Design**

- Followed these [design guidelines](https://www.behance.net/gallery/19759151/Snapscan-iOs-design-and-branding?tracking_source=), including:

- Colors.
- Typography: font face, size and weight.
- Layout: composition and space between elements.

Original design idea by [Gregoire Vella on Behance](https://www.behance.net/gregoirevella).

**Interactions**

-  Splash screen
        A simple page with the name of your app, and links to the sign up and log in pages.

-  Sign up and log in pages

  - The user is able to register in the app with full name, email and password.
  - The user can log into the app using email and password.
  -  If the user is not logged in, they can't access pages that require the user to be logged in.

-  Home page (catagories page)

  -  When the user logs in, they are presented with the categories page.
  - For each category, the user can see their name, icon and the total amount of all the transactions that belongs to that category.
  - When the user clicks (or taps) on a category item, the application navigates to the transactions page for that category.
  - There is a button "add a new category" at the bottom that brings the user to the page to create a new category.

-  Transactions page
  -  For a given category, the list of transactions is presented, ordered by the most recent.
  - At the top of the page the user could see the total amount for the category (sum of all of the amounts of the transactions in that category).
  -  There is a button "add a new transaction" at the bottom that brings the user to the page to create a new transaction.
  -  When the user clicks on the "Back" button (<), the user navigates to the home page.

-  "Add a new category" page

  - The user fills out a form to create a new category, indicating their name and icon (both mandatory).
  - The user click (or taps) the "Save" button to create the new category, and is taken to the home page on success.
  - When the user clicks on the "Back" button (<), the user navigates to the home page.

- "Add a new transaction" page

   - The user fills out a form to create a new transaction with:
     - name (mandatory)
     - amount (mandatory)
     - categories (mandatory)

   - The user click (or taps) the "Save" button to create the new transation, and is taken to the transactions page for that category.
   - When the user clicks on the "Back" button (<), the user navigates to the transactions page for that category.

**Testing**
  - Createed unit and integration tests for all the most important components of the RoR application.

**Technical requirements**

  - Used Postgres as your database.
  - Used devise for authentication.
  - Validated all user input to make sure that anyone with bad intentions cannot compromise the app.
  - The project is deployed and accessible online.
  - The database schema reflects the following structure:
<p align="center">
  image
</p>

**Video Demo**

- Click [here](https://drive.google.com/file/d/1nzuq6zkgsB7Z6hqGqdmswTAGJ8MZlpFH/view?usp=sharing)  explanation

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -96,193 +65,110 @@ After you're finished please remove all the comments and instructions!
...
-<p align="right">(<a href="#readme-top">back to top</a>)</p>
-
-<!-- LICENSE -->
-
 ## 📝 License <a name="license"></a>
 
-This project is [MIT](./LICENSE) licensed.
+This project is [MIT](./MIT.md) licensed.
 
 _NOTE: we recommend using the [MIT license](https://choosealicense.com/licenses/mit/) - you can set it up quickly by [using templates available on GitHub](https://docs.github.com/en/communities/setting-up-your-project-for-healthy-contributions/adding-a-license-to-a-repository). You can also use [any other license](https://choosealicense.com/licenses/) if you wish._
```
Comment (Whoistolu) [inline README.md]: - [ ] [OPTIONAL] Well done @AbigiyaTY with the licensing of this web app, however, I strongly advise that you remove the `_NOTE:`  part as it is just a recommendation from MIT.

File: README.md [hunk 2]
```diff
@@ -1,88 +1,57 @@
...
+    - [Setup](#setup)
+    - [Install](#install)
+    - [Usage](#usage)
+  - [Author](#author)
+  - [🔭 Future Features ](#-future-features-)
+  - [🤝 Contributing ](#-contributing-)
+  - [⭐️ Show your support ](#️-show-your-support-)
+  - [📝 License ](#-license-)
+
+# 📖 [ Budget App] <a name="about-project"></a>
```
Comment (Whoistolu) [inline README.md:23]: - [ ] [REQUIRED] There is a key part of the README that you have narrowly skipped: The Acknowledgement section". As stated by the project requirements, kindly ensure the "Acknowledgements" section gives appropriate credit to [Gregoire Vella on Behance](https://www.behance.net/gregoirevella) the [author of the original design](https://www.behance.net/gallery/19759151/Snapscan-iOs-design-and-branding?tracking_source=), as required by the [Creative Commons license of the design](https://creativecommons.org/licenses/by-nc/4.0/).

File: spec/feature/category_spec.rb [hunk 3]
```diff
@@ -0,0 +1,27 @@
+require 'rails_helper'
+
+RSpec.describe Category, type: :system do
+  subject(:category) { Category.create(name: 'Food', icon: 'https://freepngimg.com/download/burger/6-2-burger-png-image.png', user_id: 1) }
+
+  it 'should not be empty' do
+    category.name = nil
+    expect(category).to_not be_valid
+  end
```
Comment (Whoistolu) [inline spec/feature/category_spec.rb:9]: - Great job on the tests 👍🏽

**Review by Whoistolu — CHANGES_REQUESTED** [review]
Hi @AbigiyaTY 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️

- [ ]  _Check the inline comments under the review for sections to improve._

### Optional suggestions
Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better.

Cheers and Happy coding!

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @whoistolu in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: README.md (hunk 1 above)
Comment (rloterh) [reply README.md]: well resolved! ✔️

File: README.md (hunk 2 above)
Comment (rloterh) [reply README.md:23]: well resolved! ✔️

File: app/views/devise/sessions/new.html.erb [hunk 4]
```diff
@@ -0,0 +1,35 @@
+<div class= "container-fluid singup">
```
Comment (rloterh) [inline app/views/devise/sessions/new.html.erb:1]: - [ ] OPTIONAL
Your app is functioning great, however, I think we can improve further on our design to look more closer to the [template design](https://www.behance.net/gallery/19759151/Snapscan-iOs-design-and-branding?tracking_source=), so as make our app a good fit for showcasing on a portfolio. Kindly bear in mind the typography, color alignment and spacing of objects.
- You can consider improving these pages:
  * Splash screen 
  * Sign up and log in pages
  * Home page (categories page)
  * Transactions page
  * "Add a new category" page
  * "Add a new transaction" page

File: README.md [hunk 5]
```diff
@@ -96,143 +65,95 @@ After you're finished please remove all the comments and instructions!
...
 
 ### Usage
 
 To run the project, execute the following command:
 
-<!--
-Example command:
-
 ```sh
   rails server
```
Comment (rloterh) [inline README.md:130]: - [ ] OPTIONAL
Similarly, it's good idea to add instructions on how to run test cases for your app.

File: README.md [hunk 6]
```diff
@@ -241,48 +162,20 @@ Feel free to check the [issues page](../../issues/).
...
-<!-- LICENSE -->
-
 ## 📝 License <a name="license"></a>
 
-This project is [MIT](./LICENSE) licensed.
-
-_NOTE: we recommend using the [MIT license](https://choosealicense.com/licenses/mit/) - you can set it up quickly by [using templates available on GitHub](https://docs.github.com/en/communities/setting-up-your-project-for-healthy-contributions/adding-a-license-to-a-repository). You can also use [any other license](https://choosealicense.com/licenses/) if you wish._
-
-<p align="right">(<a href="#readme-top">back to top</a>)</p>
+This project is [MIT](./MIT.md) licensed.
```
Comment (rloterh) [inline README.md:181]: - [ ] OPTIONAL
MIT license link is throwing a 404 error when visited. Kindly endeavor to fix it.

**Review by rloterh — APPROVED** [review]
## Status: Approved :heavy_check_mark:

Hi Abigail,  

Your project is complete! 🏁 Nice job! There is nothing else to say other than... it's time to merge it 🍾 Congratulations!  🎉💯🌟 

## Highlights 🎯 
- Properly added required models, controllers, and corresponding views!
- Devise correctly implemented for authentication!
- Schema follow given ERD!
- App is running as intended!
- Good ReadMe!

## Optional suggestions 🧰 
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you consider them as they can improve your code._

_Kindly consider my inline comments marked **[OPTIONAL]** under this review_

**You can also consider:**
- You used incorrect git-flow/ GitHub flow for this project. Kindly note that for capstone project worked on by a single contributor should have merge request flow **from feature into development**. Kindly note this important step on subsequent projects.

Keep soaring higher! 💻 🍷 🚀 

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. Remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think the code review was unfair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**